lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -337,48 +493,1338 @@ LocaleMatch lookupMatcher(
+  //  fit
+  auto localeMatchResult = lookupMatcher(requestedLocales, availableLocales);
+  //  5. Let result be a new Record.
+  ResolveLocale result;
+  std::u16string value;
+  std::vector<std::u16string> supportedExtensionAdditionKeys;
+  //  9. For each element key of relevantExtensionKeys, do
+  for (std::u16string key : relevantExtensionKeys) {
+    result.value = u"";
+    if (localeMatchResult.extension != u"") { // 9.h.
+      if (localeMatchResult.extension.find(key)) { // 9.h.i.
+        std::u16string requestedValue = localeMatchResult.extension;
+        if (requestedValue != u"") {
+          value = requestedValue;
+        } else {
+          value = u"true";
+        }
+        supportedExtensionAdditionKeys.push_back(key);
+      }
+    }
+    if (options.find(key) != options.end()) { // 9.i.
+      auto optionsValue = options.find(key)->second;
+      if (optionsValue.isString()) {
+        if (optionsValue.getString() == u"") {
+          optionsValue = true;
+        }
+        if (optionsValue.getString() != u"" &&
+            optionsValue.getString() != value) {
+          supportedExtensionAdditionKeys.erase(
+              std::remove(
+                  supportedExtensionAdditionKeys.begin(),
+                  supportedExtensionAdditionKeys.end(),
+                  key),
+              supportedExtensionAdditionKeys.end());
+          value = optionsValue.getString();
+        }
+      }
+      if (resolveKnownAliases(key, value) != u"") {
+        if (!value.empty() && !isValidKeyword(key, value)) {
+          result.key = u"";
+          continue;
+        }
+        result.key = value;
+      }
+      for (std::u16string supportedExtendionKey :
+           supportedExtensionAdditionKeys) {
+        std::vector<std::u16string> valueList;
+        std::u16string keyValue = localeMatchResult.extension;
+        keyValue = resolveKnownAliases(localeMatchResult.extension, keyValue);
+        if (!keyValue.empty() &&
+            !isValidKeyword(supportedExtendionKey, keyValue)) {
+          continue;
+        }
+        valueList.push_back(keyValue);
+        // Void function but SHOULD affect localeMatchResult.matchedLocale?
+        setUnicodeExtensions(supportedExtendionKey, valueList);
+        localeMatchResult.matchedLocale = keyValue; // not right
+      }
+    }
+  }
+  result.locale = localeMatchResult.locale;
+  return result;
+}
+// Implementation of
+// https://tc39.es/ecma402/#sec-intl.datetimeformat.supportedlocalesof
 vm::CallResult<std::vector<std::u16string>> DateTimeFormat::supportedLocalesOf(
     vm::Runtime *runtime,
     const std::vector<std::u16string> &locales,
     const Options &options) noexcept {
-  return std::vector<std::u16string>{u"en-CA", u"de-DE"};
+  // 1. Let availableLocales be %DateTimeFormat%.[[AvailableLocales]].
+  // 2. Let requestedLocales be ? CanonicalizeLocaleList(locales).
+  vm::CallResult<std::vector<std::u16string>> requestedLocales =
```

Comment:
Can use `auto` here as well. And in general it's a good thing to use type inference as much as possible.